src/DataFramesMeta.jl

```diff
@@ -52,6 +53,88 @@ replace_syms!(e::Expr, membernames) =
         mapexpr(x -> replace_syms!(x, membernames), e)
     end
 
+is_simple_non_broadcast_call(x) = false
+function is_simple_non_broadcast_call(expr::Expr)
+    (expr.head == :call
+        && length(expr.args) >= 2
+        && expr.args[1] isa Symbol
+        && all(x -> x isa QuoteNode || onearg(x, :cols), expr.args[2:end]))
+end
+
+is_simple_broadcast_call(x) = false
+function is_simple_broadcast_call(expr::Expr)
+    (expr.head == :.
+        && length(expr.args) == 2
+        && expr.args[1] isa Symbol
+        && expr.args[2] isa Expr
+        && expr.args[2].head == :tuple
+        && all(x -> x isa QuoteNode || onearg(x, :cols), expr.args[2].args))
+end
+
+function is_simple_function_call(x)
+    is_simple_non_broadcast_call(x) || is_simple_broadcast_call(x)
+end
+
+function args_to_selectors(v)
+    t = map(v) do arg
+        if arg isa QuoteNode
+            arg
+        elseif onearg(arg, :cols)
+            arg.args[2]
+        else
+            Throw(ArgumentError("This path should not be reached, arg: $(arg)"))
+        end
+    end
+
+    :(DataFramesMeta.make_source_concrete($(Expr(:vect, t...))))
+end
+
+function get_source_fun(function_expr)
```

Comment:
Maybe add a comment explaining what this function does?

Reply:
added a docstring.